api/v1alpha1/istiorevisiontags_types.go

```diff
@@ -0,0 +1,204 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	IstioRevisionTagKind = "IstioRevisionTag"
+	DefaultRevisionTag   = "default"
+)
+
+// IstioRevisionTagSpec defines the desired state of IstioRevisionTag
+type IstioRevisionTagSpec struct {
+	// +kubebuilder:validation:Required
+	TargetRef IstioRevisionTagTargetReference `json:"targetRef"`
```

Comment:
If an `IstioRevisionTag` can reference an `Istio`, doesn't that make the name "revision tag" confusing? Should it be called `IstioTag`?

Also, why exactly do we need the option to reference `Istio`? How do the two reference options fit into the fully-automatic updates? The plan was for the operator to automatically change the `istio.io/rev` label from the old to the new revision, so it should probably also change the `targetRef` in `IstioRevisionTag`. If we decide against the latter, then we should also not implement the former, but then the automatic update wouldn't actually do anything. 

We should discuss this and go over all possible scenarios.

Reply:
Well, revision tag is the upstream term. So I'd prefer to keep it around.

The reason it can reference an Istio is exactly to make the upgrade use case easier. The operator doesn't need to change any labels for an automatic upgrade anymore, it will just update the tag to point to the new IstioRevision (it does that already). I honestly think that's the more elegant solution that changing labels on workloads and namespaces - IMO the operator shouldn't touch those user resources.